cas_client/Cargo.toml

```diff
@@ -29,16 +26,37 @@ futures = { workspace = true }
 heed = { workspace = true }
 http = { workspace = true }
 more-asserts = { workspace = true }
-reqwest = { workspace = true }
-reqwest-middleware = { workspace = true }
-reqwest-retry = { workspace = true }
 tempfile = { workspace = true }
 thiserror = { workspace = true }
 tokio = { workspace = true }
 tokio-retry = { workspace = true }
 tracing = { workspace = true }
 url = { workspace = true }
 
+reqwest-middleware = "0.4"
+reqwest-retry = "0.7"
+reqwest = { version = "0.12", features = ["json", "trust-dns", "stream"] }
```

Comment:
`trust-dns` has been re-named to `hickory-dns`: https://docs.rs/reqwest/latest/reqwest/#optional-features, https://github.com/seanmonstar/reqwest/blob/master/Cargo.toml#L74.